packages/backend-common/src/service/createPluginRouter.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Router from 'express-promise-router';
+import { Router as ExpressRouter } from 'express';
+import { notFoundHandler } from '../middleware';
+import { PluginEnvironment } from '../types';
+import { useHotMemoize } from '../hot';
+
+export type CreatePluginRouterOptions = {
```

Comment:
```suggestion
export type CreatePluginRouterOptions<Environment extends PluginEnvironment = PluginEnvironment> = {
```

etc downwards? May be that you want to put me than the prescribed stuff in there

Reply:
I "think" to go outside the standard environment, you would need from extend from PluginEnvironment with additional fields. PluginEnvironment fields are required, but the usage pattern uses destructuring to pick what the plugin needs. Assuming you can extend this the standard `makeCreateEnv` in `packages/backend/src/ndex/` it probably should be fine?